### PR TITLE
feat(metrics): identify if user is patient 0 upon login

### LINKED ISSUE #7128: Add `isPatient0` as a user property
### Acceptance Criteria (optional)
- [ ] When user connects, we make an `identify` call to Segment and include the property `isPatient0`

**Estimated effort:** X points ([see CONTRIBUTING.md](https://github.com/ParabolInc/parabol/blob/master/CONTRIBUTING.md#points-and-sizes))

### PR DESCRIPTION
**Pull request by tianrunhe**
# Description

Fixes #7128 

## Demo

Screen Shot 2022-08-29 at 4 49 18 PM

## Testing scenarios

1. Log in
2. Verify in Segment the `isPatient0` property is included in the `identify` call

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [x] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/server/graphql/mutations/helpers/bootstrapNewUser.ts [hunk 1]
```diff
@@ -15,14 +16,18 @@ import createTeamAndLeader from './createTeamAndLeader'
 
 // no waiting necessary, it's just analytics
 const handleSegment = async (user: User, isInvited: boolean) => {
-  const {id: userId, createdAt, email, segmentId, picture, preferredName} = user
+  const {id: userId, createdAt, email, featureFlags, tier, segmentId, preferredName} = user
+  const domain = email.split('@')[1]
   segmentIo.identify({
     userId,
     traits: {
-      avatar: picture,
```
Comment (tianrunhe) [inline packages/server/graphql/mutations/helpers/bootstrapNewUser.ts:22]: This removal is intentional: useless property in Amplitude

**Comment by tianrunhe** [comment]
Hey @BartoszJarocki , can you give this a look? Thanks

**Comment by BartoszJarocki** [comment]
@tianrunhe it's still a draft PR, should I still take a look?

**Comment by tianrunhe** [comment]
> @tianrunhe it's still a draft PR, should I still take a look?

Yes it's ready! We now have a couple of checks running so I forgot to press the button after they all finished 😂.

File: packages/server/postgres/queries/getPatientZeroByDomain.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import getPg from '../getPg'
+import IUser from '../types/IUser'
+import {getPatientZeroByDomainQuery} from './generated/getPatientZeroByDomainQuery'
+
+const getPatient0ByDomain = async (domain: string): Promise<IUser> => {
+  const usersInDomain = await getPatientZeroByDomainQuery.run({domain}, getPg())
+  return usersInDomain[0] as unknown as IUser
```
Comment (BartoszJarocki) [inline packages/server/postgres/queries/getPatientZeroByDomain.ts:7]: +1 no need to do `as unknown`, as `IUser` will work just fine. also, if you do that, there's no need to specify the return type, ts will infer it. so you can also remove `Promise<IUser` from l5

File: packages/server/graphql/mutations/helpers/bootstrapNewUser.ts [hunk 3]
```diff
@@ -15,14 +16,18 @@ import createTeamAndLeader from './createTeamAndLeader'
...
       createdAt,
       email,
-      name: preferredName
+      name: preferredName,
+      isActive: true,
+      featureFlags,
+      highestTier: tier,
+      isPatient0: domain ? userId === (await getPatientZeroByDomain(domain)).id : undefined
```
Comment (BartoszJarocki) [inline packages/server/graphql/mutations/helpers/bootstrapNewUser.ts:30]: +1 this is repeated in a couple of places, it'd be nice to have a helper method that'd do this

File: packages/server/postgres/queries/getPatientZeroByDomain.ts (hunk 2 above)
Comment (tianrunhe) [reply packages/server/postgres/queries/getPatientZeroByDomain.ts:7]: I got this error if not converting to `unknown` first:
Screen Shot 2022-08-30 at 10 15 56 AM
I think that's why we are doing it that way today, e.g., [here](https://github.com/ParabolInc/parabol/blob/master/packages/server/postgres/queries/getUsersByDomain.ts#L7)

Comment (BartoszJarocki) [reply packages/server/postgres/queries/getPatientZeroByDomain.ts:7]: oh interesting, it worked just fine on my end. let's leave it as is then!

**Pull request merged**